### PR TITLE
Image add support for onerror callback

### PR DESCRIPTION
**Pull request by skathuria29**
Closes Github issue # [5905](https://github.com/adobe/react-spectrum/issues/5905) 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

### CODE REVIEW

**Comment by skathuria29** [comment]
> Thanks for the addition, and test+story!
> 
> You'll need to sign the [CLA](https://github.com/adobe/react-spectrum/pull/5945/checks?check_run_id=21978046750) for the build to pass.

hey @reidbarber I'm not an external contributor. I have raised request to add company to my public GitHub profile. Waiting for approval.

File: packages/@react-spectrum/image/docs/Image.mdx
Comment (snowystinger) [inline packages/@react-spectrum/image/docs/Image.mdx:90]: I think you meant it to be the same as the Object Fit example above?
`https://i.imgur.com/c3gTKSJ.jpg`

and I think you meant the src down below to be this image, so that we go through the onError

**Review by snowystinger — COMMENTED** [review]
If we're going to support `onError` then we should also support `onLoad`, would you mind adding it?

File: packages/@react-spectrum/image/docs/Image.mdx
Comment (skathuria29) [reply packages/@react-spectrum/image/docs/Image.mdx:90]: I'm sorry i didn't understand your statement clearly. 
In this example, what I'm trying to do is to set a default image in case the image loas fails. `src` is accepted as a prop. if the `src` image url fails to load, the `DEFAULT_IMAGE` url will be set as src.

Comment (snowystinger) [reply packages/@react-spectrum/image/docs/Image.mdx:90]: The default image is currently the one that will fail. Src is set to an image which exists.

Comment (skathuria29) [reply packages/@react-spectrum/image/docs/Image.mdx]: ohh that. i have modified that in the example. please verify.

**Comment by skathuria29** [comment]
@snowystinger @reidbarber Please review the changes.

**Comment by snowystinger** [comment]
GET_BUILD

File: packages/@react-types/image/src/index.d.ts
Comment (snowystinger) [inline packages/@react-types/image/src/index.d.ts:30]:
```suggestion
   * Callback - If an error occurs while loading or rendering an image, see [Image loading errors](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#image_loading_errors).
```

File: packages/@react-types/image/src/index.d.ts
Comment (snowystinger) [inline packages/@react-types/image/src/index.d.ts:34]:
```suggestion
   * Callback - Called when the image has successfully loaded, see [load event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/load_event).
```

**Review by snowystinger — APPROVED** [review]

**Comment by snowystinger** [comment]
GET_BUILD

**Review by snowystinger — APPROVED** [review]

**Comment by skathuria29** [comment]
@reidbarber Please review the changes

**Review by reidbarber — APPROVED** [review]
LGTM

**Pull request merged**
